lib/tests/test_revset.rs

```diff
@@ -1030,6 +1031,27 @@ fn try_resolve_commit_ids(
         .collect())
 }
 
+fn try_evaluate_expression(repo: &dyn Repo, revset_str: &str) -> Result<(), RevsetEvaluationError> {
+    let settings = testutils::user_settings();
+    let context = RevsetParseContext {
+        aliases_map: &RevsetAliasesMap::default(),
+        local_variables: HashMap::new(),
+        user_email: settings.user_email(),
+        date_pattern_context: chrono::Utc::now().fixed_offset().into(),
+        extensions: &RevsetExtensions::default(),
+        workspace: None,
+    };
+    let expression = parse(&mut RevsetDiagnostics::new(), revset_str, &context).unwrap();
+    let symbol_resolver = default_symbol_resolver(repo);
+    let expression = expression
+        .resolve_user_expression(repo, &symbol_resolver)
```

Comment:
Looks like this part is shared with `try_resolve_commit_ids()`. Extract a `try_resolve_expression()`?